src/map/pc.c

```diff
@@ -9456,6 +9456,21 @@ bool pc_equipitem(struct map_session_data *sd,short n,int req_pos)
 		return false;
 	}
 
+	// dont equip arrow if weapon equipped is not bow / guitar / whip.
+	// allow assassin, baby and parent jobs to equip ammuntions without weapons (need official info)
+	if ((sd->status.weapon != W_BOW && sd->status.weapon != W_MUSICAL && sd->status.weapon != W_WHIP)
+		&& (sd->class_&MAPID_BASEMASK != MAPID_ASSASSIN || sd->class_&MAPID_BASEMASK != MAPID_ASSASSIN_CROSS || sd->class_&MAPID_BASEMASK != MAPID_BABY_ASSASSIN
+		|| sd->class_&MAPID_BASEMASK != MAPID_GUILLOTINE_CROSS || sd->class_&MAPID_BASEMASK != MAPID_GUILLOTINE_CROSS_T || sd->class_&MAPID_BASEMASK != MAPID_BABY_CROSS)){
+		if (id->type == IT_AMMO){
+			int w_idx = sd->equip_index[EQI_HAND_R];
+			enum weapon_type w_type = (w_idx != -1) ? (enum weapon_type)sd->inventory_data[w_idx]->look : W_FIST;
+			if (w_idx == -1 || (id->look != W_BOW && id->look != W_MUSICAL && id->look != W_WHIP)){
+				clif_equipitemack(sd, 0, 0, ITEM_EQUIP_ACK_FAIL);
```

Comment:
all good. altough I'm now wondering why pc_equipitem, have some direct call on clif.
I mean it's working but I feel that clif call would be an issue to use that fonction on a script or some other internal.
It's not at all in the scope of that change, but still I'm just droping this here. ;)